ygot/render.go

```diff
@@ -1249,25 +1237,41 @@ func constructJSONSlice(field reflect.Value, parentMod string, args jsonOutputCo
 // This function extracts field index 0 of the struct within the interface and returns
 // the value.
 func unionInterfaceValue(v reflect.Value, appendModuleName bool) (interface{}, error) {
-	switch {
-	case v.Kind() != reflect.Ptr && v.Kind() != reflect.Interface:
+	if !util.IsValueInterfaceToStructPtr(v) {
 		return nil, fmt.Errorf("received a union type which was invalid: %v", v.Kind())
-	case v.Elem().Kind() != reflect.Ptr:
-		return nil, fmt.Errorf("received a union type which was not a pointer: %v", v.Kind())
-	case v.Elem().Elem().Kind() != reflect.Struct:
-		return nil, fmt.Errorf("received a union type that did not contain a struct: %v", v.Kind())
-	case v.Elem().Elem().NumField() != 1:
+	}
+
+	if !util.StructValueHasNFields(v.Elem().Elem(), 1) {
```

Comment:
nit: to avoid doing .Elem().Elem() multiple times, do v = v.Elem().Elem() // now a struct

Reply:
Done.